src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,49 +1,87 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
 import {createDrawerNavigator} from '@react-navigation/drawer';
+import {withOnyx} from 'react-native-onyx';
 
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {getLastAccessedReport} from '../../reportUtils';
 import styles, {
     getNavigationDrawerType,
     getNavigationDrawerStyle,
 } from '../../../styles/styles';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../compose';
+import SCREENS from '../../../SCREENS';
 
 // Screens
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import ReportScreen from '../../../pages/home/ReportScreen';
 
 const propTypes = {
+    // Available reports that would be displayed in this navigator
+    reports: PropTypes.objectOf(PropTypes.shape({
+        reportID: PropTypes.number,
+    })),
+
     ...windowDimensionsPropTypes,
 };
 
+const defaultProps = {
+    reports: {},
+};
+
 const Drawer = createDrawerNavigator();
 
-const MainDrawerNavigator = props => (
-    <Drawer.Navigator
-        openByDefault
-        drawerType={getNavigationDrawerType(props.isSmallScreenWidth)}
-        drawerStyle={getNavigationDrawerStyle(
-            props.windowWidth,
-            props.isSmallScreenWidth,
-        )}
-        sceneContainerStyle={styles.navigationSceneContainer}
-        edgeWidth={500}
-        drawerContent={() => <SidebarScreen />}
-    >
-        <Drawer.Screen
-            name="Report"
-            component={ReportScreen}
-
-            // Providing an empty string here will ensure that the ReportScreen does not show as '/r/undefined'
-            // eslint-disable-next-line react/jsx-props-no-multi-spaces
-            initialParams={{reportID: ''}}
-            options={{
+// Decorated to always returning the result of the first call - keeps Screen initialParams from changing
```

Comment:
This comment is a little hard to follow and I might change it to something like:

>We are decorating `getLastAccessedReport` so that it caches the first result once the reports load. This will ensure the `initialParams` are only set once for the `ReportScreen`.

Which also begs the question... what happens if the `initialParams` change? And why do we care? Maybe we can explain that as well here just so it is very obvious?

Reply:
Nothing will happen if we pass different `initialParams` after the screen is mounted. And that's why we want to stop calculating them each time `MainDrawerNavigator` re-renders

It might be better to just remove this `_.once` optimization and address it if it ever becomes a problem

Reply:
Nice one, yep I think this is fine in any case but took me a second to get why so a small comment would have been helpful.